File: growing-shop-auth-before/src/main/java/com/example/growingshopauth/auth/service/PolicyService.java
package com.example.growingshopauth.auth.service;

import com.example.growingshopauth.auth.domain.Policies;
import com.example.growingshopauth.auth.domain.Policy;
import com.example.growingshopauth.auth.dto.RoleRequest;
import com.example.growingshopauth.auth.repository.PolicyRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class PolicyService {
    private final PolicyRepository policyRepository;

    public Policies findAll() {
        return new Policies(policyRepository.findAll());
    }

    @Transactional
    public Policy create(RoleRequest.CreatePolicy req) {
        return policyRepository.save(req.toEntity());
    }

    @Transactional
    public void delete(Long id) {
        policyRepository.deleteById(id);
    }
}

File: growing-shop-auth/src/main/kotlin/com/example/growingshopauth/auth/service/PolicyService.kt
package com.example.growingshopauth.auth.service

import com.example.domain.auth.Policies
import com.example.domain.auth.Policy
import com.example.growingshopauth.auth.dto.RoleRequest
import com.example.repository.auth.PolicyRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class PolicyService(
    private val policyRepository: PolicyRepository
) {

    fun findAll(): Policies = Policies(policyRepository.findAll())

    @Transactional
    fun create(req: RoleRequest.CreatePolicy): Policy = policyRepository.save(req.toEntity())

    @Transactional
    fun delete(id: Long) {
        policyRepository.deleteById(id)
    }
}
